src/test/ui/async-await/issue-70935-complex-spans.stderr

```diff
@@ -0,0 +1,31 @@
+error: future cannot be sent between threads safely
+  --> $DIR/issue-70935-complex-spans.rs:10:45
+   |
+LL |   fn foo(tx: std::sync::mpsc::Sender<i32>) -> impl Future + Send {
+   |                                               ^^^^^^^^^^^^^^^^^^ future created by async block is not `Send`
+LL |
+LL | /     async move {
+LL | |         baz(|| async{
+LL | |             foo(tx.clone());
+LL | |         }).await;
+LL | |     }
+   | |_____- this returned value is of type `impl std::future::Future`
+   |
+   = help: the trait `std::marker::Sync` is not implemented for `std::sync::mpsc::Sender<i32>`
+note: future is not `Send` as this value is used across an await
+  --> $DIR/issue-70935-complex-spans.rs:13:9
+   |
+LL |            baz(|| async{
+   |  __________^___-
+   | | _________|
+   | ||
+LL | ||             foo(tx.clone());
+LL | ||         }).await;
+   | ||         -      ^- value is later dropped here
+   | ||_________|______|
+   | |__________|      await occurs here, with value maybe used later
```

Comment:
This should be something like `the value` instead.

We should also detect these kinds of overlapping spans and emit two three separate notes instead of just one.

Reply:
Tweaked some conditions: db80f28
It should be better, though I'm not sure if it's the best.